### PR TITLE
Compatibility with scikit-learn v0.16.1

### PR DESCRIPTION
**Pull request by desilinguist**
- Update `scikit-learn` to version 0.16.1 in `.travis.yml`, `requirements.txt`, and `requirements_rtd.txt`.
- Remove SVR coefficient correction now that the original bug has been fixed in scikit-learn (#111).
- Remove dependency on `BaseLibLinear` in `model_params()` since it is no longer exposed in scikit-learn (#233, #235).
- Expose `LinearSVR` from scikit-learn and so change `SVR` to *not* have a 'linear' kernel by default.
- Include `RescaledLinearSVR` among the rescaled regressors.
- Remove `SVR` from `test_linear_models()` in `test_regression.py` and create a new `test_non_linear_models()`.
- Update one of the expected values in `test_sparse_predict_sampler()` likely due to the following item in the scikit-learn 0.16 release notes: *"RBFSampler with gamma=g formerly approximated rbf_kernel with gamma=g/2.; the definition of gamma is now consistent, which may substantially change your results if you use a fixed value."*
- Update expected values in `test_scaling()` likely due to this item in the scikit-learn v0.16 release notes: *"Fix numerical stability issues in linear_model.SGDClassifier and linear_model.SGDRegressor by  clipping large gradients and ensuring that weight decay rescaling is always positive."*

### CODE REVIEW

**Comment by desilinguist** [comment]
@dan-blanchard and @mheilman, I assigned @aoifecahill because I can only assign one person to a PR but can you guys also take a look, if you can spare the time?

**Comment by desilinguist** [comment]
I am not going to spend any more time worrying about the coverage here since (a) I can't figure out what exactly what lines have are not covered anymore (coveralls.io shows all lines that are not covered, not just the ones that lost coverage), and (b) because the decrease is so small.

**Comment by aoifecahill** [comment]
don't you want to merge into develop?

**Comment by desilinguist** [comment]
This is skll, remember? master = develop and stable = master.

**Comment by aoifecahill** [comment]
Yep, sorry. Realized that too late :)

**Comment by aoifecahill** [comment]
Looks fine to me :)
